src/test/java/bio/terra/service/snapshotbuilder/SnapshotRequestDaoTest.java

```diff
@@ -40,23 +41,33 @@ class SnapshotRequestDaoTest {
   @Autowired private SnapshotRequestDao snapshotRequestDao;
 
   private Dataset dataset;
+  private Snapshot snapshot;
   private SnapshotAccessRequest snapshotAccessRequest;
 
   private static final String EMAIL = "user@gmail.com";
 
   @BeforeEach
   void beforeEach() throws IOException {
     dataset = daoOperations.createDataset(DaoOperations.DATASET_MINIMAL);
+    snapshot = daoOperations.createAndIngestSnapshot(dataset, DaoOperations.SNAPSHOT_MINIMAL);
     snapshotAccessRequest = SnapshotBuilderTestData.createSnapshotAccessRequest();
   }
 
   private SnapshotAccessRequestResponse createRequest() {
     return snapshotRequestDao.create(dataset.getId(), snapshotAccessRequest, EMAIL);
   }
 
-  private void verifyResponseContents(SnapshotAccessRequestResponse response) {
+  private SnapshotAccessRequestResponse createRequestV2() {
+    return snapshotRequestDao.createV2(snapshot.getId(), snapshotAccessRequest, EMAIL);
+  }
+
+  private void verifyResponseContents(SnapshotAccessRequestResponse response, boolean isV2) {
```

Comment:
I added a isV2 variable to this, to check the snapshot id instead of the dataset id. Again, this will be removed with follow on work to more completely to snapshot ids, but there may be a better solution in the mean time.

Reply:
I would go the other way here and either remove the dataset support completely, or rename the datatset tests as "old" and write the new ones using the names that they will have once the dataset code is removed.

My goal would be to make any follow-on work to remove dataset as easy as possible and require little to no change of the new snapshot code.

Reply:
Yeah, I like the idea the instead renaming the dataset endpoints to "_old".